elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/ShadedClassLoader.java

```diff
@@ -116,8 +171,8 @@ public String getPackageName(String className) {
         return null;
     }
 
-    private byte[] getClassBytes(String name) throws ClassNotFoundException {
-        try (InputStream is = getResourceAsStream(name.replace('.', '/') + CLASS_EXTENSION)) {
+    private byte[] getShadedClassBytes(String name) throws ClassNotFoundException {
```

Comment:
Can there ever be a case where this receives a `name` that is already shaded via `getShadedResourceName`?

Reply:
I don't see how. This is part of the `loadClass` execution path, where `name` is the name of the class to be loaded. Since `agent` is not part of the actual package name and since we do not relocate classes, noone should look for a class that starts with `agent.`.
`getShadedResourceName` is only called within the `getResource` path, where `name` refers to the file resource, rather than the class name.